core/src/avm2/events.rs

```diff
@@ -448,10 +453,33 @@ pub fn dispatch_event<'gc>(
         parent = parent_of(par);
     }
 
+    let dispatched = event.as_event().unwrap().dispatched;
+
+    let event = if dispatched {
+        event
+            .call_public_property("clone", &[], activation)?
+            .coerce_to_object(activation)
```

Comment:
nit: I think `as_object` should be a little faster here, and not require passing down an Activation. You can do `unwrap_or_else` instead of the `map_err`, too.